### PR TITLE
[MGDOBR-348] Use resource classes for rest calls

### PR DESCRIPTION
**Pull request by radtriste**
[MGDOBR-348](https://issues.redhat.com/browse/MGDOBR-348) 

Issue splitted into different PRs for better review:
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/364
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/365
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/366
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/367
- https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/368

Depends on https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/pull/365

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/common/Utils.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.redhat.service.bridge.integration.tests.common;
+
+public class Utils {
+
+    public static String getSystemProperty(String parameters) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/common/Utils.java:5]: Is it needed to keep the method in separate class? It seems to be used just in `BridgeUtils.java`

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/BridgeResource.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    public static BridgeListResponse getBridgeList(String token) {
+        return getBridgeListResponse(token)
+                .then()
+                .statusCode(200)
+                .extract()
+                .as(BridgeListResponse.class);
+    }
+
+    public static void getBridgeListFails(int failStatusCode) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/BridgeResource.java:45]: Maybe to change to `getBridgeListResponse` and do the status code check as part of step implementation?

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/IngressResource.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    public static Response postJsonEventResponse(String token, String endpoint, InputStream cloudEventStream, Headers headers) {
+        return ResourceUtils.jsonRequest(token)
+                .headers(headers)
+                .body(cloudEventStream)
+                .contentType(ContentType.JSON)
+                .post(endpoint + "/events");
+    }
+
+    public static Response postJsonEventToFullURLResponse(String token, String url, InputStream cloudEventStream, Headers headers) {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/IngressResource.java:29]: Is this needed? I think the functionality can be achieved by using `postJsonEventResponse` with some non existing endpoint parameter.

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/ResourceUtils.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Optional;
+
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+
+import static io.restassured.RestAssured.given;
+
+public class ResourceUtils {
+
+    public static RequestSpecification jsonRequest() {
```
Comment (sutaakar) [inline integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/ResourceUtils.java:12]: Method seems unused.

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/common/Utils.java (hunk 1 above)
Comment (sutaakar) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/common/Utils.java:5]: I see, there will be more functionality added in next PRs.

Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/common/Utils.java:5]: It was just in foresee of the next changes ... I will set it back to `BridgeUtils`

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/ResourceUtils.java (hunk 4 above)
Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/ResourceUtils.java:12]: removed

File: integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/BridgeResource.java (hunk 2 above)
Comment (radtriste) [reply integration-tests/src/test/java/com/redhat/service/bridge/integration/tests/resources/BridgeResource.java:45]: removed

**Review by sutaakar — APPROVED** [review]

**Review by jiripetrlik — APPROVED** [review]

**Review by Srihari1192 — APPROVED** [review]

**Pull request merged**